config/nunjucks-component-extension.js

```diff
@@ -0,0 +1,39 @@
+const nunjucks = require('nunjucks');
+const changeCase = require('change-case');
+const logger = require('../lib/logger');
+
+function ComponentExtension() {
+  this.tags = ['component'];
+
+  this.parse = function parse(parser, nodes) {
+    const tok = parser.nextToken();
+    const args = parser.parseSignature(null, true);
+
+    parser.advanceAfterBlockEnd(tok.value);
+
+    return new nodes.CallExtension(this, 'run', args);
+  };
+
+  this.run = function run(context, data, args) {
+    const name = typeof args === 'undefined' ? data.name : data;
+    const ctx = typeof args === 'undefined' ? data.context : args;
+
+    let result = '';
+
+    try {
+      result = nunjucks.render(`_components/${changeCase.paramCase(name)}.nunjucks`, ctx);
```

Comment:
Yeah, probably worth doing as we might move to the preferred extension of `.njk` soon.